dragonchain/job_processor/job_processor.py

```diff
@@ -105,7 +117,7 @@ def get_next_task() -> Optional[dict]:
         The next task. Blocks until a job is found.
     """
     _log.info("Awaiting contract task...")
-    pop_result = redis.brpop_sync("mq:contract-task", 0, decode=False)
+    pop_result = redis.brpoplpush_sync(CONTRACT_TASK_KEY, PENDING_TASK_KEY, 0, decode=False)
     if pop_result is None:
         return None
```

Comment:
Anywhere this function returns `None` (intentionally throwing away the task), you need to remove the pending task that you just added (since we're _trying_ to intentionally ignore that task)

Reply:
if `pop_result` is None, doesn't that mean the redis command failed?

Reply:
`pop_result` being none means the blocking wait timed out (so yeah, i guess  you could call that a failure)

Reply:
if the blocking wait times out, won't that also mean nothing gets added to the second queue?

Reply:
I guess that's true for this particular branch. so i guess remove the lpop_sync from this branch, but later in this method, there is another `return None` which does need to be fixed